### PR TITLE
141 forwarded ref all components

### PR DESCRIPTION
**Pull request by wmangimelli**

### CODE REVIEW

File: packages/hs-react-ui/README.md [hunk 1]
```diff
@@ -233,6 +233,16 @@ import { Div } from 'path/to/htmlElements';
...
+
+You might be wondering, 'Why don't I just pass a "ref" property along with the sub-component's Props?" And, this in fact would work...most of the time. However, there are some internal "ref"-ing that goes on behind the scenes and at those occurrences passing the ref inside the sub-component props would not work. For example, if you want to reference the Card's container element, create a ref, perhaps with React.createRef<HTMLDivElement>(), and pass that ref to the _containerRef_ prop, **not** through the _containerProps_ as {ref: myRef}.
+
+Since all the Foundry components are functional components, it was determined that refForwarding wasn't necessary and wouldn't be supported. So, don't pass a "ref" prop to a Foundry component because it won't be forwarded. Instead, simply create a ref and explicitly pass it to the sub-component you want to target using the supported API props.
+
```
Comment (aVileBroker) [inline packages/hs-react-ui/README.md:245]: Love this write-up!

File: packages/hs-react-ui/src/components/Button/Button.stories.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ storiesOf('Button', module)
         <Button
           variant={select('variant', variants, variants.fill)}
           color={color('color', colors.primaryDark)}
-          onClick={action('button-click')}
+          onClick={() => action('button-click')()}
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Button/Button.stories.tsx:42]: is this necessary? action() returns a function which should get called by onClick - here you're passing a function which fires the function returned from action(). It's a weird case but I feel like the old version should have been correct

File: packages/hs-react-ui/src/utils/refs.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+/**
+ * Merges multiple references
+ * @param {Array} refs - Array of reference object or reference functions
+ * @return {RefCallback} To apply each ref whether it is a ref object or ref callback
+ */
+export type MergeRefs = Array<React.MutableRefObject<any> | React.RefCallback<any> | undefined>;
+
+export const mergeRefs = (refs: MergeRefs): null | React.RefCallback<any> => {
+  if (!(refs instanceof Array)) return null;
+  const filteredRefs = refs.filter(Boolean);
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/utils/refs.ts:12]: this is so clean!

File: packages/hs-react-ui/src/components/Card/Card.stories.tsx [hunk 4]
```diff
@@ -97,4 +97,37 @@ storiesOf('Card', module)
...
+          elevation={number('elevation', 2, { range: true, min: -5, max: 5, step: 1 })}
+          onClick={boolean('onClick', true) ? e => onClick(e) : undefined}
+          feedbackType={select('feedbackType', feedbackTypes, feedbackTypes.ripple)}
+          containerRef={cardContainerRef}
+          headerRef={cardHeaderRef}
+          interactiveFeedbackRef={interactiveFeedbackRef}
+        >
+          {text(
+            'children',
+            'A Hello, World! program generally is a computer program that outputs or displays the message Hello, World!.',
```
[the remark this answers is not in the document]
Comment (wmangimelli) [reply packages/hs-react-ui/src/components/Card/Card.stories.tsx:127]: Ah, yes. Good call!

File: packages/hs-react-ui/src/components/Button/Button.stories.tsx (hunk 2 above)
Comment (wmangimelli) [reply packages/hs-react-ui/src/components/Button/Button.stories.tsx:42]: Yep, you are right. Switching it back.

**Review by aVileBroker — APPROVED** [review]

**Pull request merged**